<?xml version="1.0"?>
<launch>
<!-- ===============================================================
     = A) Global variables                                         =
     ===============================================================   -->

   <arg name="AreaSizeX" default="61" />
   <arg name="AreaSizeY" default="21" />
   <arg name="AreaSizeZ" default="6" />
   <arg name="CellSize" default="1" />
   <arg name="SourcePositionX" default="5" />
   <arg name="SourcePositionY" default="15" />
   <arg name="SourcePositionZ"  default="2" />
   <arg name="FixedFrame" default="/map" />
   <!-- ethanol=0; methane=1; hydrogen=2; propanol=3; chlorine=4; flurorine=5; acetone=6; neon=7; helium=8; hot air=9-->
   <arg name="GasType" default="0" />
   
   <!--node pkg="tf" type="static_transform_publisher" name="sens1_broadcaster" args="7 10 3 0 0 0 map sensor01_frame 100" /-->

<node pkg="gas_concentration_visualization" type="gas_concentration_visualization" name="gas_concentration_visualization" output="screen">
		<param name="/MIN_X" value="0.0" />
		<param name="/MAX_X" value="61.0" />
		<param name="/MIN_Y" value="0.0" />
		<param name="/MAX_Y" value="21.0" />
		<param name="/MIN_Z" value="0.0" />
		<param name="/MAX_Z" value="6.0" />
		<param name="/CELL_SIZE" value="$(arg CellSize)" />
		<param name="/COLORMAP" value="jet" />
		<param name="/MAX_CONCENTRATION" value="0" />
		<param name="/MIN_CONCENTRATION" value="0" />
		<param name="/FIXED_FRAME" value="$(arg FixedFrame)" />
		<param name="/NUM_SNAPSHOTS" value="500" />
        <param name="/NUM_N_POINTS_MAP" value="100" />
        <param name="/result_location" value="/home/han/gdm_simulator_catkin_ws/src/gas_dispersion_simulation/recordedData/results/gasType0_simulation" />
		<param name="/VALUE" value="cVal" />
   </node>
   <!-- ===============================================================
     = C) Environment node                                         =
     ===============================================================   -->
   <node pkg="environment" type="environment" name="environment">
	    <param name="/cell_size" value="$(arg CellSize)"/>
	    <param name="/area_size_x" value="$(arg AreaSizeX)"/>
	    <param name="/area_size_y" value="$(arg AreaSizeY)"/>
	    <param name="/area_size_z" value="$(arg AreaSizeZ)"/>
	    <param name="/source_position_x" value="$(arg SourcePositionX)"/>
	    <param name="/source_position_y" value="$(arg SourcePositionY)"/>
	    <param name="/source_position_z" value="$(arg SourcePositionZ)"/>
	    <param name="/fixed_frame" value="$(arg FixedFrame)"/>
	    <param name="/environment_data" value="/home/han/gdm_simulator_catkin_ws/data/environments/original.env" />
   </node>

   <node pkg="rqt_rviz" type="rqt_rviz" name="rqt_rviz" />
   
   <!--node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 world map 100" /-->
</launch>
